web3/contract.py

```diff
@@ -122,6 +172,8 @@ class Contract(object):
     clone_bin = None
 
     dev_doc = None
+    function = None
```

Comment:
*naming nitpick*

I think this an `events` should be pluralized.

Reply:
I like that these are all exactly the same as the ABI type field right now (which is why the written issue had them singular as well). At the same time I get the model. I proposed one more option for naming back on #419. It would require a significant change, but hopefully address this naming issue.